sympy/calculus/util.py

```diff
@@ -112,3 +119,672 @@ def elm_domain(expr, intrvl):
+                    if other.min.is_negative:
+                        if other.max.is_zero:
+                            return Limits(-oo, self.min/other.min)
+                        if other.max.is_positive:
+                            # the actual answer is a Union of Limits,
+                            # Union(Limits(-oo, self.min/other.min),
+                            #       Limits(self.min/other.max, oo))
+                            return Limits(-oo, oo)
+
+                    if other.min.is_zero and other.max.is_positive:
+                        return Limits(self.min/other.max, oo)
+
+            elif other.is_real:
+                if other is S.Infinity or other is S.NegativeInfinity:
+                    if self == Limits(-oo, oo):
+                        return Limits(-oo, oo)
+                    if self.max is S.Infinity:
+                        return Limits(Min(0, other), Max(0, other))
+                    if self.min is S.NegativeInfinity:
+                        return Limits(Min(0, -other), Max(0, -other))
+                if other.is_positive:
+                    return Limits(self.min/other, self.max/other)
+                elif other.is_negative:
+                    return Limits(self.max/other, self.min/other)
+            return Mul(self, 1/other, evaluate=False)
+
+        return NotImplemented
+
+    __truediv__ = __div__
+
+    @_sympifyit('other', NotImplemented)
+    def __rdiv__(self, other):
+        if isinstance(other, Expr):
+            if other.is_real:
+                if other.is_zero:
+                    return S.Zero
+                if S.Zero in self:
+                    if self.min == S.Zero:
+                        if other.is_positive:
+                            return Limits(Mul(other, 1/self.max), oo)
+                        if other.is_negative:
+                            return Limits(-oo, Mul(other, 1/self.max))
+                    if self.max == S.Zero:
+                        if other.is_positive:
+                            return Limits(-oo, Mul(other, 1/self.min))
+                        if other.is_negative:
+                            return Limits(Mul(other, 1/self.min), oo)
+                    return Limits(-oo, oo)
+                else:
+                    return Limits(Min(other/self.min, other/self.max),
+                            Max(other/self.min, other/self.max))
+            return Mul(other, 1/self, evaluate=False)
+        else:
+            return NotImplemented
+
+    __rtruediv__ = __rdiv__
+
+    @_sympifyit('other', NotImplemented)
+    def __pow__(self, other):
+        from sympy.functions.elementary.miscellaneous import real_root
+        if isinstance(other, Expr):
+            if other is S.Infinity:
+                if self.min.is_nonnegative:
+                    if self.max < 1:
+                        return S.Zero
+                    if self.max == 1:
+                        return FiniteSet(0, 1)
```

Comment:
I am beginning to think that not even ``Limits(0. 1)`` would cover all possibilities (we could have max > 1 in the result).
It might be best to consider only three cases: max < 1 (as above), min > 1 (as below) and else return ``Limits(0, oo)``. The cases with max or min equal to 1 could give almost any results.